### DISCUSSION TITLE
USWDS - Question: Mobile header - open a section accordion to current page link

### DISCUSSION THREAD

**Discussion by ajanickiv** [uswds/uswds#6932] [Q&A]
### Is your feature request related to a problem? Please describe.

This is just a question, but there is no question issue type :)

On the USWDS documentation site, when in mobile header mode, the top level section accordion buttons open and shows the link that maps the current page automatically.
![image](https://github.com/uswds/uswds/assets/16595489/f6325509-dcfc-41b0-8580-b1f8d449eb40)

Is this functionality built into the header or is there additional logic running that performs that automatic opening? When we implemented the header, it does not automatically open the section accordions to the current page link.

Wondering if we set up our header incorrectly or if we should add some JS logic to handle it.

### Describe the solution you'd like

N/A

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Feature+Request%22) for duplicate feature requests.

**Comment by ajanickiv** [comment]
Just checking in to see if there is any answer to my query :)

**Comment by ryparker** [comment]
I'm closing this during the backlog review. As I take over maintenance, I'm trying to make the open list reflect work the team can realistically take on. This has been quiet, with no work underway.

If you still need help with it, please reply here and I'll reopen it. You don't need to file another issue. A current example would help us understand where you're running into the problem.

**Comment by ajanickiv** [comment]
I am really just looking to understand how the functionality I described is implemented as we would like similar functionality in our app. On the USWDS documentation site, when in mobile header mode, **the top level section accordion buttons open and shows the link that maps the current page automatically.** Any help would be appreciated.

**Comment by ryparker** [comment]
Thanks for following up, and sorry we left your original question unanswered. I've reopened this as promised and classified it as a question.

The documentation site adds this behavior in its [navigation template](https://github.com/uswds/uswds-site/blob/main/_includes/nav/page-collapsible.html#L1-L9). It checks whether the current page belongs to a navigation section and renders that section's button with `aria-expanded="true"`. USWDS's [accordion JavaScript](https://github.com/uswds/uswds/blob/develop/packages/usa-accordion/src/index.js#L84-L88) then opens the associated submenu during initialization. The header does not automatically detect the current route.

To reproduce this, have your app's template or routing logic set `aria-expanded="true"` on the current section's button before USWDS initializes, and `"false"` on the others. Keep `aria-controls` pointing to the submenu's ID. Adding `usa-current` alone does not expand the section.

The docs site uses [separate mobile and desktop navigation markup](https://github.com/uswds/uswds-site/blob/main/_includes/navbar.html#L21-L36). If your app shares markup between both layouts, the initial expanded state will apply to both unless you add responsive handling.

Does that clarify how to implement it in your app? If your navigation updates without a full page load, let us know how you're rendering it so we can help with updating the expanded state after route changes.

**Comment by ryparker** [comment]
I've moved this to Q&A with the original question and replies, including the implementation details above. Please continue here if you have follow-up questions; no new issue is needed.
